pkg/asset/agent/image/agentartifacts.go

```diff
@@ -75,15 +81,24 @@ func (a *AgentArtifacts) Generate(_ context.Context, dependencies asset.Parents)
 	a.ISOPath = baseIso.File.Filename
 	a.Kargs = kargs.KernelCmdLine()
 
-	if agentconfig.Config != nil {
-		a.BootArtifactsBaseURL = strings.Trim(agentconfig.Config.BootArtifactsBaseURL, "/")
-		// External platform will always create a minimal ISO
-		a.MinimalISO = agentconfig.Config.MinimalISO || agentManifests.AgentClusterInstall.Spec.PlatformType == hiveext.ExternalPlatformType
-		if agentconfig.Config.MinimalISO {
-			logrus.Infof("Minimal ISO will be created based on configuration")
-		} else if agentManifests.AgentClusterInstall.Spec.PlatformType == hiveext.ExternalPlatformType {
-			logrus.Infof("Minimal ISO will be created for External platform")
+	switch agentWorkflow.Workflow {
+	case workflow.AgentWorkflowTypeInstall:
+		if agentconfig.Config != nil {
+			a.BootArtifactsBaseURL = strings.Trim(agentconfig.Config.BootArtifactsBaseURL, "/")
+			// External platform will always create a minimal ISO
+			a.MinimalISO = agentconfig.Config.MinimalISO || agentManifests.AgentClusterInstall.Spec.PlatformType == hiveext.ExternalPlatformType
+			if agentconfig.Config.MinimalISO {
+				logrus.Infof("Minimal ISO will be created based on configuration")
+			} else if agentManifests.AgentClusterInstall.Spec.PlatformType == hiveext.ExternalPlatformType {
+				logrus.Infof("Minimal ISO will be created for External platform")
+			}
+		}
+	case workflow.AgentWorkflowTypeAddNodes:
+		if addNodesConfig.Config.BootArtifactsBaseURL != "" {
```

Comment:
From a maintenance point of view, I think it would be useful to centralize all the relevant configs just in the `ClusterInfo` asset, leaving out as much as possible the `AddNodesConfig` from the asset graph.
In another PR I followed the same principle, see the [retrieveAdditionalNTPSources()](https://github.com/openshift/installer/pull/9108/files#diff-b6c0ec51beec87580161335112657421a6874afa61790f8300295cbae1a90c96) method